core/primitives/src/state.rs

```diff
@@ -110,3 +110,14 @@ impl FlatStateValue {
         }
     }
 }
+
+/// Value to insert into the trie.
+#[derive(BorshSerialize, BorshDeserialize, Debug, Clone, PartialEq, Eq, ProtocolSchema)]
+pub enum ValueToInsert {
+    /// Full value, works for all possible operations, triggers both memtrie
+    /// and disk changes generation.
+    Full(Vec<u8>),
```

Comment:
We could store the hash to avoid double computations, but I like more this data layout.

Reply:
I want to fix it here, see https://github.com/near/nearcore/pull/12344#issuecomment-2448537785